trader/trader.go

```diff
@@ -121,15 +124,18 @@ func (t *Trader) Start() {
 		currentUpdateTime := time.Now()
 		if lastUpdateTime.IsZero() || t.timeController.ShouldUpdate(lastUpdateTime, currentUpdateTime) {
 			success := t.update()
-			millisForUpdate := time.Since(currentUpdateTime).Milliseconds()
-			e := t.threadTracker.TriggerGoroutine(func(inputs []interface{}) {
-				e := t.metricsTracker.SendUpdateEvent(currentUpdateTime, success, millisForUpdate)
+			if shouldSendUpdateMetric(t.startTime, currentUpdateTime, t.metricsTracker.LastMetricUpdateTime) {
+				millisForUpdate := time.Since(currentUpdateTime).Milliseconds()
+				e := t.threadTracker.TriggerGoroutine(func(inputs []interface{}) {
+					e := t.metricsTracker.SendUpdateEvent(currentUpdateTime, success, millisForUpdate)
+					if e != nil {
+						log.Printf("failed to send update event metric: %s", e)
+					}
+					t.metricsTracker.LastMetricUpdateTime = currentUpdateTime
```

Comment:
I think I may not have communicated my intent very clearly here. My bad.

We don’t want to update the time stamp here, especially not as an exported variable. Instead, we want the metricsTracker to update it on itself in the sendUpdsteEvent function on success.